### PR TITLE
AP_OSD: get wind speed from wind vane on rover

### PR DESCRIPTION
**Pull request by IamPete1**
This allows plotting of apparent wind from the wind vane on rover. 

Because its possible to have wind direction but not speed, there is a little funny business to get the arrow to draw with a short vector. There is a nice long comment explaining this.

image

### CODE REVIEW

File: libraries/AP_OSD/AP_OSD_Screen.cpp [hunk 1]
```diff
@@ -1188,11 +1191,11 @@ void AP_OSD_Screen::draw_message(uint8_t x, uint8_t y)
     }
 }
 
-void AP_OSD_Screen::draw_speed_vector(uint8_t x, uint8_t y,Vector2f v, int32_t yaw)
+void AP_OSD_Screen::draw_speed_vector(uint8_t x, uint8_t y,Vector2f v, int32_t yaw, bool force_arrow)
```
Comment (tridge) [inline libraries/AP_OSD/AP_OSD_Screen.cpp:1194]: change this to be a function that takes a body frame vector, and add a comment explaining what it does

**Review by tridge — CHANGES_REQUESTED** [review]

File: libraries/AP_OSD/AP_OSD_Screen.cpp (hunk 1 above)
Comment (IamPete1) [reply libraries/AP_OSD/AP_OSD_Screen.cpp]: Moved to a function that takes a angle and magnitude

File: libraries/AP_OSD/AP_OSD_Screen.cpp [hunk 2]
```diff
@@ -1227,7 +1227,14 @@ void AP_OSD_Screen::draw_gspeed(uint8_t x, uint8_t y)
     WITH_SEMAPHORE(ahrs.get_semaphore());
     Vector2f v = ahrs.groundspeed_vector();
     backend->write(x, y, false, "%c", SYM_GSPD);
-    draw_speed_vector(x + 1, y, v, ahrs.yaw_sensor);
+
+    float angle = 0;
+    const float lenght = v.length();
```
Comment (tridge) [inline libraries/AP_OSD/AP_OSD_Screen.cpp:1232]: length please

File: libraries/AP_OSD/AP_OSD_Screen.cpp [hunk 3]
```diff
@@ -1367,14 +1374,28 @@ void AP_OSD_Screen::draw_compass(uint8_t x, uint8_t y)
...
 void AP_OSD_Screen::draw_wind(uint8_t x, uint8_t y)
 {
+#if !APM_BUILD_TYPE(APM_BUILD_Rover)
     AP_AHRS &ahrs = AP::ahrs();
     WITH_SEMAPHORE(ahrs.get_semaphore());
     Vector3f v = ahrs.wind_estimate();
-    if (check_option(AP_OSD::OPTION_INVERTED_WIND)) {
-        v = -v;
+    float angle = 0;
+    const float lenght = v.length();
```
Comment (tridge) [inline libraries/AP_OSD/AP_OSD_Screen.cpp:1382]: length here too

**Review by tridge — CHANGES_REQUESTED** [review]

**Comment by tridge** [comment]
ok to merge once spelling fixed :-)

File: libraries/AP_OSD/AP_OSD_Screen.cpp (hunk 3 above)
Comment (IamPete1) [reply libraries/AP_OSD/AP_OSD_Screen.cpp]: fixed

File: libraries/AP_OSD/AP_OSD_Screen.cpp (hunk 2 above)
Comment (IamPete1) [reply libraries/AP_OSD/AP_OSD_Screen.cpp]: fixed

**Review by tridge — APPROVED** [review]

**Pull request merged**